### PR TITLE
[feature](merge-on-write) enable merge-on-write by default again

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

fix #27188, #28096

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run feut

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

File: regression-test/suites/update/test_update_unique.groovy [hunk 1]
```diff
@@ -29,7 +29,7 @@ suite("test_update_unique", "p0") {
                 date_value date
             )
             UNIQUE KEY(k)
-            DISTRIBUTED BY HASH(k) BUCKETS 5 properties("replication_num" = "1");
+            DISTRIBUTED BY HASH(k) BUCKETS 5 properties("replication_num" = "1", "enable_unique_key_merge_on_write" = "false");
```
Comment (zhannngchen) [inline regression-test/suites/update/test_update_unique.groovy:32]: Why MoW can't work here?

File: regression-test/suites/schema_change_p0/test_schema_change.groovy [hunk 2]
```diff
@@ -41,7 +41,7 @@ suite("test_schema_change") {
                 PARTITION p201707 VALUES LESS THAN ('2021-12-01')
             )
             DISTRIBUTED BY HASH(siteid) BUCKETS 5
-            PROPERTIES("replication_num" = "1", "light_schema_change" = "true"); 
+            PROPERTIES("replication_num" = "1", "light_schema_change" = "true", "enable_unique_key_merge_on_write" = "false");
```
Comment (zhannngchen) [inline regression-test/suites/schema_change_p0/test_schema_change.groovy:44]: ditto

File: regression-test/suites/schema_change_p0/test_rename_column.groovy [hunk 3]
```diff
@@ -41,7 +41,7 @@ suite ("test_rename_column") {
             `min_dwell_time` INT DEFAULT "99999" COMMENT "用户最小停留时间")
         UNIQUE KEY(`user_id`, `date`, `city`, `age`, `sex`) DISTRIBUTED BY HASH(`user_id`)
         BUCKETS 8
-        PROPERTIES ( "replication_num" = "1" , "light_schema_change" = "false")
+        PROPERTIES ( "replication_num" = "1" , "light_schema_change" = "false", "enable_unique_key_merge_on_write" = "false")
```
Comment (zhannngchen) [inline regression-test/suites/schema_change_p0/test_rename_column.groovy:44]: ditto

**Review by zhannngchen — CHANGES_REQUESTED** [review]
For each case that must use MOR, please add a comment to explain the reason.

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

File: regression-test/suites/datatype_p0/datev1/test_datev1_common.groovy [hunk 4]
```diff
@@ -371,12 +371,14 @@ suite("test_datev1_common", "nonConcurrent") {
         """)
     }
 
+    sql """set delete_without_partition=true; """
```
Comment (liaoxin01) [inline regression-test/suites/datatype_p0/datev1/test_datev1_common.groovy:374]: Is the delete partition behavior inconsistent between mow and mor？

Comment (bobhan1) [reply regression-test/suites/datatype_p0/datev1/test_datev1_common.groovy:374]: 从文档的描述来看似乎是mor表的行为有问题，这里删除条件推断不出来分区 https://doris.apache.org/zh-CN/docs/dev/sql-manual/sql-reference/Data-Manipulation-Statements/Manipulation/DELETE/
> 语法一中，如果为分区表，需要指定分区，如果不指定，doris 会从条件中推断出分区。两种情况下，doris 无法从条件中推断出分区: 1) 条件中不包含分区列；2) 分区列的 op 为 not in。当分区表未指定分区，或者无法从条件中推断分区的时候，需要设置会话变量 delete_without_partition 为 true，此时 delete 会应用到所有分区。

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run external

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**